Dockerfile

```diff
@@ -0,0 +1,34 @@
+# One Dockerfile to rule them all
+# Example:
+# docker build --build-arg APP=nexus -t keeper:latest --target distroless-base .
+FROM golang:1.23.3 AS base-builder
+WORKDIR /workspace
+COPY go.mod go.sum ./
+RUN go mod download
+
+FROM golang:1.23.3 AS builder
+WORKDIR /workspace
+COPY --from=base-builder /go/pkg/mod /go/pkg/mod
+COPY . .
+
+ARG APP
+ARG TARGETARCH
+ENV GOOS=linux \
+    GOARCH=$TARGETARCH
+RUN if [ "nexus" = "$APP" ]; then \
+        export GOEXPERIMENT=boringcrypto && \
+        export CGO_ENABLED=1; \
+    else \
+        export CGO_ENABLED=0; \
+    fi
+
+RUN echo "Building $APP for architecture: $TARGETARCH"
+RUN go build -o $APP /workspace/app/$APP/cmd/main.go
+
+FROM gcr.io/distroless/base AS distroless-base
+COPY --from=builder /workspace/$APP /binary
+ENTRYPOINT ["/binary"]
+
+FROM gcr.io/distroless/static AS distroless-static
+COPY --from=builder /workspace/$APP /binary
+ENTRYPOINT ["/binary"]
```

Comment:
Since we have `CGO_ENABLED` (at least for Nexus), we will need a mac (virtual) machine to create all cross-platform builds... (otherwise, for example a debian builder will not find relevant tool-chain to link mac/darwin bindings to build the binaries for the target os)

which is fine with me, I already have one mac build vm.

it just means, we'll need 9 (3 mac arm, 3 linux arm, 3 linux x86) container registries, instead of 3 -- which is fine with me, since there'll be some "form" of release automation anyway.

Reply:
Upon further discussion on other channels, there may be hope :)

I'll test things out and get back here :).